recipes/libjxl/all/conanfile.py

```diff
@@ -14,114 +21,215 @@ class LibjxlConan(ConanFile):
     homepage = "https://github.com/libjxl/libjxl"
     topics = ("image", "jpeg-xl", "jxl", "jpeg")
 
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
-
-    exports_sources = "CMakeLists.txt", "patches/**"
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "avx512": [True, False],
+        "avx512_spr": [True, False],
+        "avx512_zen4": [True, False],
+        "with_tcmalloc": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "avx512": False,
+        "avx512_spr": False,
+        "avx512_zen4": False,
+        "with_tcmalloc": False,
+    }
+
+    def export_sources(self):
+        copy(self, "conan_deps.cmake", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.settings.arch not in ["x86", "x86_64"] or Version(self.version) < "0.9":
+            del self.options.avx512
+            del self.options.avx512_spr
+            del self.options.avx512_zen4
+        # https://github.com/libjxl/libjxl/blob/v0.9.1/CMakeLists.txt#L52-L59
+        if self.settings.os in ["Linux", "FreeBSD"] and self.settings.arch == "x86_64":
+            self.options.with_tcmalloc = True
```

Comment:
Sorry for reviving this but I just got a surprise when updating this package with allocator mismatches due to this change in my static build and I just wanted to ask; Should we really do this here? From the linked cmake I gather that libjxl tries to use tcmalloc by default if it exists, but if we don't add it as a dependency, it shouldn't exists unless it is available on the system? Which I assume is a common problem in general with many libs. Could a better solution be to be able to disable it completely upstream instead either via a patch or a PR to them?

Reply:
Thanks for reporting this. Please open it as a separate issue for more visibility.